api/src/main/java/jakarta/config/ConfigMapping.java

```diff
@@ -55,4 +55,34 @@
      * @return a {@link String} representation of a configuration path.
      */
     String path() default "";
+
+    /**
+     * Override the configuration member name.
+     */
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ElementType.METHOD })
+    @interface Name {
+        /**
+         * The name of the configuration member name. Must not be empty.
+         *
+         * @return the configuration member name
+         */
+        String value();
+    }
+
+    /**
+     * Specify the default value of a configuration member.
+     */
+    @Documented
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.METHOD })
+    @interface Default {
```

Comment:
I'm wondering if we should provide specialized Default types like `@DefaultInt`, `@DefaultBoolean`, etc, or as a member `intValue()`, `booleanValue()`. The annotation / member to use has to match the declared type. If another annotation / member is used we could throw a validation error.

This would avoid having to come up with a Conversion strategy from String to something and provide type safety for defaults definition

Reply:
You cannot create an extra annotation for each java Type. Rather, the Default value can be a class of data converter. While we have not discussed the converters yet, it looks like it should be discussed before continuing with the `@Default` annotation.